src/service_aggregator.py

```diff
@@ -283,9 +307,9 @@ async def check_for_messages(guid, pydantic_kgraph, accumulated_results, num_que
                     async with message.process():
                         num_responses += 1
                         logger.debug(f'{guid}: Strider returned {num_responses} out of {num_queries}.')
-                        jr = await process_message(message)
-                        if await is_end_message(jr):
-                            logger.debug('Received complete message from multistrider')
+                        jr = process_message(message)
```

Comment:
Style wise I'm not sure what the right thing to do in async python. Should you make everything an async function, even if you don't think it needs to be? What if I put some blocking call in that function in the future? I lean the way you have it now, but I can see it both ways.